cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/compute/GcpInstanceResourceBuilder.java

```diff
@@ -209,12 +212,35 @@
             Operation operation = insert.execute();
             verifyOperation(operation, buildableResource);
             updateDiskSetWithInstanceName(auth, computeResources, instance);
+            assignToInstanceGroup(context, group, instance, buildableResource);
             return singletonList(createOperationAwareCloudResource(buildableResource.get(0), operation));
         } catch (GoogleJsonResponseException e) {
             throw new GcpResourceException(checkException(e), resourceType(), buildableResource.get(0).getName());
         }
     }
 
+    private void assignToInstanceGroup(GcpContext context,
```

Comment:
Can you add a javadoc about what exactly this method is doing? This method also needs more logging and comments in general. There's no explanation for anything being done right now.